### DISCUSSION TITLE
Best approach for database scaling?

### DISCUSSION THREAD

**Discussion by vladimir-stornest** [mastodon/mastodon#16413] [Q&A]
I'm interested in scaling the database with master-slave replication using Makra, as explained in the [official documentation](https://docs.joinmastodon.org/admin/scaling/).

As I understand, when one of the servers fails, Makara will blacklist it so it won't send traffic to it. But what happens if the master server fails? Then the Mastodon instance won't work because it won't be able to do write operations. 

I assume I need to install and configure _repmgr_ for automated failover to promote the slave server to master. How will the Mastadon instance know that there is a new master server when the old server will still be hardcoded in the configuration file? Could pgBouncer help with this, and how?

Any feedback is appreciated!

**Comment by nightpool** [comment]
Managing failures of the primary database is out of scope of the Mastodon app. Read replicas, as we use them, are designed to increase the amount of traffic that Mastodon can handle, and do nothing to handle failures in the main database process.
